archaius-core/src/main/java/com/netflix/config/DynamicBooleanProperty.java

```diff
@@ -23,14 +23,41 @@
  *
  */
 public class DynamicBooleanProperty extends PropertyWrapper<Boolean> {
+
+    protected volatile boolean primitiveValue;
+
     public DynamicBooleanProperty(String propName, boolean defaultValue) {
         super(propName, Boolean.valueOf(defaultValue));
+
+        // Set the initial value of the cached primitive value.
+        this.primitiveValue = chooseValue();
+
+        // Add a callback to update the cached primitive value when the property is changed.
+        this.prop.addCallback(new Runnable() {
```

Comment:
Yes that is a much better way of doing it. 
@howardyuan - i've made this change to the PR. Does this satisfy your concerns around my previous usage of the callbacks?